packages/react-components/src/components/TroubleShootingGuideErrorBar.tsx

```diff
@@ -0,0 +1,150 @@
+    camera: PermissionState;
+    microphone: PermissionState;
+  }) => void;
+  /**
+   * Optional callback to supply users with further troubleshooting steps for network issues
+   * experienced when connecting to a call.
+   *
+   * @example
+   * ```ts
+   * onNetworkingTroubleShootingClick?: () =>
+   *  window.open('https://contoso.com/network-troubleshooting', '_blank');
+   * ```
+   *
+   * @remarks
+   * if this is not supplied, the composite will not show a 'network troubleshooting' link.
+   */
+  onNetworkingTroubleShootingClick?: () => void;
+  /**
+   * strings related to trouble shooting guidance link and dismiss button
+   */
+  troubleShootingGuideStrings: _TroubleShootingGuideErrorBarStrings;
+}
+
+/**
+ * @internal
+ * A component to show device Permission/network connection related errors, contains link that leads to further trouble shooting guide
+ */
+export const _TroubleShootingGuideErrorBar = (props: _TroubleShootingGuideErrorBarProps): JSX.Element => {
+  const theme = useTheme();
+  // error bar strings
+  const localeStrings = useLocale().strings.errorBar;
+  const strings = props.strings ?? localeStrings;
+
+  const [dismissedErrors, setDismissedErrors] = useState<DismissedError[]>([]);
+
+  const {
+    onPermissionsTroubleshootingClick,
+    onNetworkingTroubleShootingClick,
+    permissionsState,
+    troubleShootingGuideStrings
+  } = props;
+
+  // dropDismissalsForInactiveErrors only returns a new object if `dismissedErrors` actually changes.
+  // Without this behaviour, this `useEffect` block would cause a render loop.
+  useEffect(
+    () => setDismissedErrors(dropDismissalsForInactiveErrors(props.activeErrorMessages, dismissedErrors)),
+    [props.activeErrorMessages, dismissedErrors]
+  );
+
+  const toShow = errorsToShow(props.activeErrorMessages, dismissedErrors);
+
+  return (
+    <Stack data-ui-id="error-bar-stack">
+      {toShow.map((error) => (
+        <MessageBar
+          {...props}
+          styles={messageBarStyle(theme, messageBarType(error.type))}
+          key={error.type}
+          messageBarType={messageBarType(error.type)}
+          messageBarIconProps={messageBarIconProps(error.type)}
+        >
+          <Stack horizontal horizontalAlign="space-between" verticalAlign="center">
+            {onPermissionsTroubleshootingClick || onNetworkingTroubleShootingClick ? (
+              <div>
+                {strings[error.type]}
+                <Link
+                  styles={linkStyle(theme)}
+                  onClick={() => {
+                    if (onPermissionsTroubleshootingClick) {
+                      onPermissionsTroubleshootingClick(permissionsState);
+                    } else if (onNetworkingTroubleShootingClick) {
+                      onNetworkingTroubleShootingClick();
+                    }
+                  }}
+                  underline
+                >
+                  <span>{troubleShootingGuideStrings.linkText}</span>
```

Comment:
Does this need to be in a span?

Reply:
It's in a span so I can apply styling targeted to it, otherwise some how styling does not get applied to innertext